flow/layers/image_filter_layer.cc

```diff
@@ -16,25 +18,29 @@ void ImageFilterLayer::Preroll(PrerollContext* context,
   Layer::AutoPrerollSaveLayerState save =
       Layer::AutoPrerollSaveLayerState::Create(context);
 
-  child_paint_bounds_ = SkRect::MakeEmpty();
-  PrerollChildren(context, matrix, &child_paint_bounds_);
+  SkRect child_bounds = SkRect::MakeEmpty();
+  PrerollChildren(context, matrix, &child_bounds);
   if (filter_) {
-    const SkIRect filter_input_bounds = child_paint_bounds_.roundOut();
+    const SkIRect filter_input_bounds = child_bounds.roundOut();
     SkIRect filter_output_bounds =
         filter_->filterBounds(filter_input_bounds, SkMatrix::I(),
                               SkImageFilter::kForward_MapDirection);
-    set_paint_bounds(SkRect::Make(filter_output_bounds));
-  } else {
-    set_paint_bounds(child_paint_bounds_);
+    child_bounds = SkRect::Make(filter_output_bounds);
   }
+  set_paint_bounds(child_bounds);
 
   if (!context->has_platform_view && context->raster_cache &&
       SkRect::Intersects(context->cull_rect, paint_bounds())) {
     SkMatrix ctm = matrix;
 #ifndef SUPPORT_FRACTIONAL_TRANSLATION
     ctm = RasterCache::GetIntegralTransCTM(ctm);
 #endif
-    context->raster_cache->Prepare(context, this, ctm);
+    if (render_count_ >= MINIMUM_FILTER_CACHE) {
```

Comment:
I didn't do this (for now?) mainly because this is an either/or situation - either prepare the layer itself for caching or prepare the cacheable child for the alternate strategy. Such a mechanism would have to have some thought on whether or not the prepare failed due to minimum render count or whether it failed because of some other intrinsic property (transform not cacheable, layer too small or too big, etc.)

Rather than solve that issue now, I think I'm going to stick with the manual test here in the Preroll method and we can come back around to a general minimum frame count later.